Wrappers/Python/test/test_dataexample.py

```diff
@@ -158,70 +159,95 @@ class TestRemoteData(unittest.TestCase):
     def setUp(self):
         
         self.data_list = ['WALNUT','USB','KORN','SANDSTONE']
-        self.tmp_file = 'tmp.txt'
-        self.tmp_zip = 'tmp.zip'
-        with ZipFile(self.tmp_zip, 'w') as zipped_file:
-            zipped_file.writestr(self.tmp_file, np.array([1, 2, 3]))
-        with open(self.tmp_zip, 'rb') as zipped_file:
-            self.zipped_bytes = zipped_file.read()        
-    
-    def tearDown(self):
-        for data in self.data_list:
-            test_func = getattr(dataexample, data)
-            if os.path.exists(os.path.join(test_func.FOLDER)):
-                shutil.rmtree(test_func.FOLDER)
-        
-        if os.path.exists(self.tmp_zip):
-            os.remove(self.tmp_zip)
 
-        if os.path.exists(self.tmp_file):
-            os.remove(self.tmp_file)
-
-    def mock_urlopen(self, mock_urlopen):
+    def mock_urlopen(self, mock_urlopen, zipped_bytes):
         mock_response = MagicMock()
-        mock_response.read.return_value = self.zipped_bytes
+        mock_response.read.return_value = zipped_bytes
         mock_response.__enter__.return_value = mock_response
         mock_urlopen.return_value = mock_response
 
     @patch('cil.utilities.dataexample.urlopen')
     def test_unzip_remote_data(self, mock_urlopen):
-        self.mock_urlopen(mock_urlopen)
-        dataexample.REMOTEDATA._download_and_extract_from_url('.')
-        self.assertTrue(os.path.isfile(self.tmp_file))
+        
+        shapes_path = os.path.join(dataexample.CILDATA.data_dir, dataexample.TestData.SHAPES)
+
+        with NamedTemporaryFile(suffix = '.zip') as tf:
```

Comment:
can you add a comment to this code block just saying that you are creating a temporary test zip file to test the function on